lib/OpenQA/CLI/api.pm

```diff
@@ -36,32 +42,61 @@ sub command ($self, @args) {
 
     my $params;
     my @data;
-    if ($hooksrc) {
-	# if $hooksrc is a json  
-        my $hookdata = path("hook/$hooksrc.json")->slurp;
-	$data = ($hookdata);
-	my $hookparams = decode_json($data);
-	my $argsparams = $self->parse_params(\@args, \@param_file);
-	$params = {%$hookparams, %$argsparams};
-	# otherwise try to run a script
-	# here
-    } else {
-	$data = path($data_file)->slurp if $data_file;
-	@data = ($data);
-	$params = $form ? decode_json($data) : $self->parse_params(\@args, \@param_file);
+    if ($prehooksrc) {
+
+	$self->_run_script ($_) for path($posthooksrc)->list->grep(qr/post_/)->each;
+	#my $hookdata = path($prehooksrc)->slurp;
+	#$data = ($hookdata);
+	#my $hookparams = decode_json($data);
+	#my $argsparams = $self->parse_params(\@args, \@param_file);
+	#$params = {%$hookparams, %$argsparams};
     }
+    
+    $data = path($data_file)->slurp if $data_file;
+    @data = ($data);
+    $params = $form ? decode_json($data) : $self->parse_params(\@args, \@param_file);
+    use Data::Dumper;
     @data = (form => $params) if keys %$params;
-
+    print Dumper(@data[1]);
+    if ($posthooksrc) {
+	use Data::Dumper;
+	# say for path('/home/sri/myapp')->list->each;
+	# run_script check exec permissions
+	$self->_run_script ($_, $data[1]) for path($posthooksrc)->list->grep(qr/post_/)->each;
+	say '_run_script returns ' . Dumper @data;
+    }
+    die "we got to the end";
     my $headers = $self->parse_headers(@headers);
     $headers->{Accept} //= 'application/json';
     $headers->{'Content-Type'} = 'application/json' if $json;
-
     my $url = $self->url_for($path);
     my $client = $self->client($url);
     my $tx = $client->build_tx($method, $url, $headers, @data);
     $self->retry_tx($client, $tx, $retries);
 }
 
+
+sub _run_script ($self, $hook_path, $data) {
+    my $hook_fl = basename($hook_path);
+    return unless ($hook_fl =~ /^(pre|post)_/);
+    my $info    = stat($hook_path) or die "path doesnt exist $hook_path : $!";
+    unless (-x $info) {
+	log_info $_ . ' is not executable';
+	return;
+    }
+    log_info $_ . ' is called';
+    eval {
+	print 'TST ';
+	# https://stackoverflow.com/a/54500161/1462096
+	#my @out = qx($hook_path);
```

Comment:
I'm not a fan of `IPC::Run`. You may use it but you can still also just keep using Perl built-ins, e.g. like I did in one of our scripts: https://github.com/os-autoinst/scripts/blob/b4f88b41c51ef5571e3f96764a99acd43252243c/openqa-clone-and-monitor-job-from-pr#L154

Reply:
That's IPC::Run3 though (I guss you meant to comment 3 lines below?)

Reply:
Why not pick one implementation and run with it? (also to remove the dependency on `Mojo::IOLoop::ReadWriteProcess` perhaps moving that `_run_cmd_capturing_output` (and similar ones) can be moved to a single place in some other PR. 

Looks like the same thing is being implemented over and over...